backup.rb

```diff
@@ -66,41 +71,56 @@ def get_images!
     @boards.each do |board|
       print "[#{board[:name]}] ".green + "found "
       visit PINTEREST + board[:path]
-      load_all_content!
-      board[:image_urls] = js <<-JS
-        $('.pin').map(function(){ return $(this).data('closeup-url') });
-      JS
+      sleep 3
+
+      image_urls = get_unique_links!
+
+      # change links to big size images of max width 736px 
+      board[:image_urls] = image_urls.map! {|link| link.gsub('236x', '736x')}
       puts "#{board[:image_urls].size} images"
+
       download_images! board
     end
   end
 
+  def get_unique_links!
+    image_links = []
+
+      until scrolled_to_bottom?
+        load_all_content!
+
+        image_links_chunk = js <<-JS
+          $('.pinUiImage img').map(function(){ return $(this).attr('src') });
+        JS
+
+        image_links.concat image_links_chunk
+      end
+
+      image_links.uniq
+  end
+
   def download_images! board
     path = @root + 'pinterest.com' + board[:path][1..-1]
     image_urls = board[:image_urls].map(&:to_s).map(&:inspect).join(' ')
     path.mkpath
     return if image_urls.empty?
     command = <<-SH
-      cd #{path.to_s.inspect} && wget --quiet --background #{image_urls}
+      cd #{path.to_s.inspect} && wget -c --background #{image_urls}
     SH
-    # puts command
     system command
   end
 
   def load_all_content!
-    while !scrolled_to_bottom?
```

Comment:
everything works. This loop is changed to:  
  until scrolled_to_bottom?
    load_all_content!
    ...
  end
in get_unique_links! method